App.tsx

```diff
@@ -94,13 +107,20 @@ class App extends Component<Props, State> {
 
     this.unsubscribeNetListener = NetInfo.addEventListener(this.handleNetworkChange)
 
-    registerCarModule(this.onConnect, this.onDisconnect)
+    // iOS CarPlay
+    Platform.OS === 'ios' && registerCarModule(this.onConnect, this.onDisconnect)
   }
 
   componentWillUnmount() {
     this.unsubscribeNetListener && this.unsubscribeNetListener()
 
-    unregisterCarModule(this.onConnect, this.onDisconnect);
+    // iOS CarPlay
+    Platform.OS === 'ios' && unregisterCarModule(this.onConnect, this.onDisconnect)
+    // Android Auto
+    if (Platform.OS === 'android') {
+      PVEventEmitter.removeListener(PV.Events.QUEUE_HAS_UPDATED, handleAndroidAutoQueueUpdate)
+      PVEventEmitter.removeListener(PV.Events.APP_FINISHED_INITALIZING_FOR_CARPLAY, onAppInitialized)
+    }
```

Comment:
Maybe these can go in the `unregisterCarModule` function in the PVCarplay.android.ts file. Same reason as before, for more consistency :)